### PR TITLE
[profiling] display a warning when changing runtime settings

### PR DESCRIPTION
**Pull request by vickenty**
### What does this PR do?

Add a warning when a go runtime setting is modified in a running agent with internal profiling enabled. 

Increase API server timeout, as `config set internal_profiling false` may block if a cpu profile is being collected, for up to duration of the profile (15s).

### Motivation

Internal profiling library will not pick up changes in go runtime settings, and profiling session needs to be restarted to add new profile types.

### Describe how to test your changes

Start the agent with internal profiling enabled: `DD_INTERNAL_PROFILING_ENABLED=true`

Change runtime profiling settings, and check that the warning message is displayed:

```
agent config set runtime_block_profile_rate 1
agent config set runtime_mutex_profile_fraction 1
```

Restart profiling:

```
agent config set internal_profiling restart
```

Wait 10 minutes, make sure block and mutex profiles are available in the app.

### Checklist

- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] The appropriate `team/..` label has been applied, if known.
- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] The [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated if applicable.

### CODE REVIEW

File: pkg/config/settings/runtime_setting_block_profile_rate.go [hunk 1]
```diff
@@ -38,8 +38,10 @@ func (r RuntimeBlockProfileRate) Set(value interface{}) error {
 		return err
 	}
 
+	err = checkProfilingNeedsRestart(profiling.GetBlockProfileRate(), rate)
```
Comment (djmitche) [inline pkg/config/settings/runtime_setting_block_profile_rate.go:41]: The name `checkSomething..` suggests returning a boolean.  I see it's called twice, though, so it makes sense to write the error message once.  Maybe `verifyRunningProfileConfig`?

**Review by djmitche — APPROVED** [review]
I assume this is an internal feature, so needs minimal external documentation?

**Review by gh123man — APPROVED** [review]
NIT: Could you add your commit comment: 
> config set internal_profiling false` may block if a cpu profile is
being collected, for up to duration of the profile (15s)

To the PR description so it doesn't get lost? (maybe rewording to call out the why the timeout was adjusted)

**Review by alai97 — APPROVED** [review]
Looks good for docs!
